document-store/src/main/java/org/hypertrace/core/documentstore/postgres/PostgresDatastore.java

```diff
@@ -0,0 +1,135 @@
+package org.hypertrace.core.documentstore.postgres;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.mongodb.ConnectionString;
+import com.typesafe.config.Config;
+import org.hypertrace.core.documentstore.Collection;
+import org.hypertrace.core.documentstore.Datastore;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.sql.*;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+import static org.hypertrace.core.documentstore.postgres.PostgresCollection.*;
+
+public class PostgresDatastore implements Datastore {
+  
+  private static final Logger LOGGER = LoggerFactory.getLogger(PostgresDatastore.class);
+  
+  private Connection client;
+  // Specifies whether document will be stored in json/jsonb format.
+  private String database = "postgres";
+  
+  @Override
+  public boolean init(Config config) {
+    try {
+      DriverManager.registerDriver(new org.postgresql.Driver());
+      String url;
+      if (config.hasPath("url")) {
+        url = config.getString("url");
+      } else {
+        String hostName = config.getString("host");
+        int port = config.getInt("port");
+        url = String.format("jdbc:postgresql://%s:%s/", hostName, port);
```

Comment:
@jainnidhi703 I have added this one as well to read based on host/port properties.